### DISCUSSION TITLE
Applying `static` modifier to any expression

### DISCUSSION THREAD

**Discussion by alrz** [dotnet/csharplang#8912] [Language Ideas]
While `field` keyword removes the need to declare a separate field for properties, this would do the same for a static field where it's being used only once within the container type.

```cs
class C {  
  private readonly static s_cache = SomeMethod();
  private readonly static SomeType Data = new(() => s_cache);
}
```
```cs
class C {  
  private readonly static SomeType Data = new(() => static StaticMethod());
}
```
`static` expressions would have the same limitations defined by static lambdas.

**Comment by TahirAhmadov** [comment]
#832

**Reply by alrz** [reply]
Static variables would not entail any constraint on the value. In comparison, this would simplify codegen to a greater degree by just moving the expression to the containing type static ctor. For example `AcceptsDelegate(static SomeMethod)` would ensure that the delegate can be cached in a static field and therefore must be static.

**Reply by 333fred** [reply]
I can't say I particularly like this. First, the syntax: the use of `static` here is completely different to any previous use of `static` as far as I can tell, and it immediately makes me think of `static` local functions or `static` lambdas. Second, afaict, this is just #832 with a bit of sugar, and still has the same issues that need to be resolved around threading.

**Reply by TahirAhmadov** [reply]
The threading can be solved with either nested static class static fields, or with `Lazy<T>`. 
Regarding syntax, I'm personally not stuck on the keyword `static`; I haven't heard of any other suggestions, we can always introduce a new keyword, but that would be an expensive proposition. 
I really hope #832 happens, and specifically this "inline" static expressions would be extremely helpful.

**Reply by 333fred** [reply]
> The threading can be solved with either nested static class static fields, or with Lazy<T>.

That is only true if the desired resolution is "I want it to be thread safe", which does not seem to be universally true. #832 is currently unchampioned, and barring some major shifts in the LDM view on the scenario, I do not think that it, or this idea, would be likely to happen.
